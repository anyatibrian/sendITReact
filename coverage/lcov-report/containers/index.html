<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> containers
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">98.77% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>80/81</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.74% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>18/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">95% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>19/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>78/78</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="homeContainer.js"><a href="homeContainer.js.html">homeContainer.js</a></td>
	<td data-value="96" class="pic high"><div class="chart"><div class="cover-fill" style="width: 96%;"></div><div class="cover-empty" style="width:4%;"></div></div></td>
	<td data-value="96" class="pct high">96%</td>
	<td data-value="25" class="abs high">24/25</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	</tr>

<tr>
	<td class="file high" data-value="loginContainer.js"><a href="loginContainer.js.html">loginContainer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file high" data-value="registerContainer.js"><a href="registerContainer.js.html">registerContainer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="35" class="abs high">35/35</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed May 29 2019 00:48:58 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
<script src="../block-navigation.js"></script>
</body>
</html>
