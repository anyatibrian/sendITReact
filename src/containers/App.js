import React, {Component} from "react";
import Store from "../Store";
import {ToastContainer} from "react-toastify";
import {Provider} from "react-redux";
import RouterComponent from "../components/routerComponent";
import 'react-toastify/dist/ReactToastify.css';
class App extends Component {
  render() {
    return (
      <div>
        <Provider store={Store}>
          <ToastContainer/>
          <RouterComponent />
        </Provider>
      </div>
    );
  }
}
export default App;