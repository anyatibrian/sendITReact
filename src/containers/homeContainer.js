import React,{Component} from "react";
import {connect} from "react-redux";
import {createParcel} from "../actions/createParcelActions"
import "../sass/home.scss"
import HomeComponent from "../components/homeComponent";
import ParcelListComponent from "../components/parcelListComponent";
import FooterComponent from "../components/footerComponent";
class HomeContainer extends Component{
  constructor(props) {
    super(props);
    this.state={
      display:"",
      receiversName:"",
      description:"",
      pickup:"",
      destination:"",
      weight:""
    };
  };
  toggleButton = (event) => {
    event.preventDefault();
    this.setState({display:"block"});
    if (this.state.display==="block"){
      this.setState({display:"none"});
    }
  };
  onChange=event=>{
    event.preventDefault();
    const {name, value} = event.target;
    this.setState({[name]:value});
  };
  onSubmit=(event)=>{
    event.preventDefault();
    const{createParcel} = this.props;
    const {
      receiversName,
      description,
      pickup,
      destination,
      weight
  } = this.state;
    const data = {
      receivers_name:receiversName,
      destination:destination,
      description:description,
      pickup:pickup,
      weight:parseInt(weight)

    };
    console.log(data);
    createParcel(data);
  };
  render() {
    const {
      display,
      receiversName,
      description,
      pickup,
      destination,
      weight
    } = this.state;
    return (
        <div>
          <HomeComponent
              toggleButton={this.toggleButton}
              display={display}
              receiversName={receiversName}
              description={description}
              pickup={pickup}
              destination={destination}
              onChange={this.onChange}
              weight={weight}
              onSubmit={this.onSubmit}
          />
          <ParcelListComponent/>
          <FooterComponent/>
        </div>
    );
  }
}
export const mapStateToProps = state=>({
  created:state.parcel.created,
  isLoading:state.parcel.isLoading,
  creationError:state.parcel.creationError
});
export  default connect(mapStateToProps, {createParcel})(HomeContainer)