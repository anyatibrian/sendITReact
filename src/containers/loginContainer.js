import React, { Component } from "react";
import {connect} from "react-redux";
import {loginUser} from "../actions/authActions";
import { LoginComponent } from "../components/loginComponent";
import "../sass/login.scss";

export class LoginContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username:"",
      password:""
    };
  }
  componentWillReceiveProps(nextProps) {
    if(nextProps.loginRedirect===true){
      const {history} = this.props;
      history.push("/");
    }
  }

  handleChange=event=>{
    event.preventDefault();
    const {name, value}= event.target;
    this.setState({[name]:value});
  };
  handleSubmit=event=>{
    event.preventDefault();
    const {username, password } = this.state;
    const {loginUser}= this.props;
    const data = {
      username:username,
      password:password
    };
    loginUser(data);
  };
  render() {
    const { username, password } = this.state;
    //sessionStorage.setItem("token",this.props.loginSuccess["access-token"]);
    const {isLoading} = this.props;
    return (
      <div className="">
        <LoginComponent
            username={username}
            password={password}
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            isLoading={isLoading}
        />
      </div>
    );
  }
}
export const mapStatesToProps=state=>({
  isLoading:state.login.isLoading,
  loginSuccess:state.login.loginSuccess,
  loginError:state.login.loginError,
  loginRedirect:state.login.loginRedirect
});
export default connect(mapStatesToProps, {loginUser}) (LoginContainer);
