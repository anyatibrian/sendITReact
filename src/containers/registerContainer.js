import React, { Component } from "react";
import RegisterComponent from "../components/registerComponent";
import {connect} from "react-redux";
import {createAccount} from "../actions/authActions";
import {validEmailRegex} from "../utilites/Regexs";
import "../sass/login.scss";

export class RegisterContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      confirmPassword: "",
      email: "",
      errors: {
        username: "",
        email: "",
        password: "",
        confirmPassword: ""
      },
      hidden: true,

    };
  }
  componentWillReceiveProps = (nextProps)=>{
    if(nextProps.registerRedirect===true){
      const {history}= this.props;
      history.push("/login");
    }
  };
  onChange = e => {
    e.preventDefault();
    const { name, value } = e.target;
    let { errors } = this.state;

    switch (name) {
      case "username":
        errors.username =
          value.length < 6
            ? "your username should be atleast six characters"
            : "";
        break;
      case "email":
        errors.email = !validEmailRegex.test(value)
          ? "invalid email address"
          : "";
        break;
      case "password":
        errors.password =
          value.length < 6
            ? "your password must be atleast six charactors"
            : "";
        break;
      case "confirmPassword":
        if (value !== this.state.password) {
          errors.confirmPassword = "your passwords does not match";
        } else {
          errors.confirmPassword = "";
        }
        break;
      default:
        break;
    }
    this.setState({ errors, [name]: value });
  };
  onSubmit = e => {
    e.preventDefault();
      const {createAccount} = this.props;
      const {username,password, email} = this.state;
      const data = {
        username:username,
        email:email,
        password:password
      };
    createAccount(data);
  };
  onToggle = () => {
    this.setState({ hidden: !this.state.hidden });
  };
  render() {
    const {
      username,
      password,
      email,
      confirmPassword,
      hidden,
      errors
    } = this.state;
    const {isLoading} = this.props;
    return (
      <div>
        <RegisterComponent
          username={username}
          password={password}
          email={email}
          confirmPassword={confirmPassword}
          onChange={this.onChange}
          onSubmit={this.onSubmit}
          hidden={hidden}
          onToggle={this.onToggle}
          errors={errors}
          isLoading={isLoading}
        />
      </div>
    );
  }
}
export const mapStateToProps =state=>({
  signUPSuccess: state.register.signUPSuccess,
  signUpError:state.register.signUpError,
  registerRedirect:state.register.registerRedirect,
  isLoading:state.register.isLoading
});
export default connect(mapStateToProps, {createAccount}) (RegisterContainer);
