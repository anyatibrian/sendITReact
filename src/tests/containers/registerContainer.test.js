import React from "react";
import {shallow} from "enzyme";
import {RegisterContainer, mapStateToProps} from "../../containers/registerContainer";

describe('Register container',  ()=>{
  let wrapper;
  let instance;
  let props={
    username:"",
    email:"",
    password:"",
    errors:{
      username:"",
      email:"",
      password: ""
    },
    hidden:true,
    onSubmit: jest.fn(),
    onChange: jest.fn(),
    onToggle:jest.fn(),
    createAccount:jest.fn(),
    history:{push:jest.fn()},
    registerRedirect:false
  };
  beforeEach(()=>{
    wrapper = shallow(<RegisterContainer {...props}/>);
    instance = wrapper.instance()
  });
  it('should render without crashing ', ()=>{
    expect(wrapper).toMatchSnapshot();
  });
  it('should test onchange method for wrong username', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"username",
        value:"anya"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.username).
    toEqual("your username should be atleast six characters")
  });
  it('should test onchange method for correct username', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"username",
        value:"anyatibrian"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.username).
    toEqual("")
  });
  it('should test onchange method for wrong email', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"email",
        value:"anyatibrian."
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.email).
    toEqual("invalid email address");
  });
  it('should test onchange method for correct email', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"email",
        value:"anyatibrian@gmail.com"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.email).
    toEqual("");
  });
  it('should test onchange method for wrong password', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"password",
        value:"133"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.password).
    toEqual("your password must be atleast six charactors");
  });
  it('should test onchange method for correct password', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"password",
        value:"13356788"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.password).
    toEqual("");
  });
  it('should test onchange method for match password error', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"confirmPassword",
        value:"1234567890"
      }
    };
    instance.onChange(event);
    instance.setState({password:"1234"});
    expect(instance.state.errors.confirmPassword).
    toEqual("your passwords does not match");
  });
  it('should test onchange method for match password', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"confirmPassword",
        value:"1234567890"
      }
    };
    instance.setState({password:"1234567890"});
    instance.onChange(event);
    expect(instance.state.errors.confirmPassword).
    toEqual("");
  });
  it('should test default state method for match password', ()=>{
    const event = {
      preventDefault:jest.fn(),
      target:{
        name:"defaultnow",
        value:"1234567890"
      }
    };
    instance.onChange(event);
    expect(instance.state.errors.confirmPassword).
    toEqual("");
  });
  it('should test onsubmit method of the container', ()=>{
    const event = {
      preventDefault:jest.fn()
    };
    instance.onSubmit(event);
    expect(props.createAccount).toHaveBeenCalled();
  });
  it('should test ontoggle method of the container', ()=>{
    instance.onToggle();
    expect(instance.state.hidden).toEqual(false);
  });
  it('should test redirect on  method of the container', ()=>{
    wrapper.setProps({registerRedirect:true});
   expect(props.history.push).toHaveBeenCalledWith('/login');
  });
  it('should test fail redirect   method of the container', ()=>{
    wrapper.setProps({registerRedirect:false});
    expect(props.history.push).toHaveBeenCalledTimes(0);
  });
  it('should map state to props', ()=> {
    const state={
      register:{
        isLoading:true
      }
    };
    expect(mapStateToProps(state)).toEqual({isLoading:true})
  });
});