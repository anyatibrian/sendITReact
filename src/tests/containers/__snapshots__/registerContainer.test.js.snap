// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register container should render without crashing  1`] = `ShallowWrapper {}`;
